ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -1242,22 +1242,17 @@ extension SignalProducerType {
 
 			if shouldStartUnderlyingProducer {
 				self.producer
-					.takeUntil(token.willDeallocSignal)
+					.takeUntil(tracker.deallocSignal)
 					.start(initializedObserver)
 			}
 		}
 	}
 }
 
-private extension NSObject {
-	var willDeallocSignal: SignalProducer<(), NoError> {
-		return self
-			.rac_willDeallocSignal()
-			.toSignalProducer()
-			.map { _ in () }
-			.mapError { error in
-				fatalError("Unexpected error: \(error)")
-				()
-			}
+private final class Tracker {
+	let (deallocSignal, observer) = Signal<(), NoError>.pipe()
+
```

Comment:
They are not private to the `SignalProducer`. Otherwise you couldn't do this `tracker.deallocSignal`.